File: 0965-unique-email-addresses/0965-unique-email-addresses.java
class Solution {
    public int numUniqueEmails(String[] emails) {
        HashSet<String> st = new HashSet<>();
        for(String email:emails){
            String[] parts = email.split("@");
            String local = parts[0];
            String domain = parts[1];
            if(local.contains("+")){
            local = local.substring(0, local.indexOf('+'));
            }
            local = local.replace(".", "");
            st.add(local + '@' + domain);
        }
        return st.size();
    }
}

File: 0965-unique-email-addresses/0965-unique-email-addresses.py
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        st = set()
        for email in emails:
            local, domain = email.split('@')
            name = local.split('+')[0].replace('.','')
            st.add(name + '@' + domain)
        return len(st)
        
        # for email in emails:
        #     local, domain = email.split('@')
        #     if '+' in local:
        #         local = local[:local.index('+')]
        #     if '.' in local:
        #         local = local.replace('.','')
        #     clean = local+ '@' + domain
        #     st.add(clean)
        # return len(st)
